### ISSUE TITLE
Linter reporting flow types as undefined

### ISSUE THREAD

**Issue by murphyrandle** [babel/babel-eslint#130]
I just upgraded babel-eslint to 3.1.15 and babel to 5.5.6.

It appears that babel-eslint no longer ignores flow types. I'm getting 'no-undef' and 'camelcase' errors for all of the types throughout my codebase.

I'd post console output, but I can't imagine how it would be helpful to you.
Any ideas on how to move forward? Are any of you seeing this behavior? My co-workers are reporting the same thing on their machines.

**Comment by sebmck** [comment]
It's correctly linting flow types, why are you expecting them to be ignored?

**Comment by murphyrandle** [comment]
Sorry, here's a contrived example: 

```javascript
export function example(course: Course): boolean {
  return !isFantastic(course)
}
```

eslint throws the error: `error "Course" is not defined`, as if it were an undefined variable. Really, it's a flow type that is defined in an external "types" file.

Does that make more sense?

**Comment by murphyrandle** [comment]
Very cool, doing some more testing and I see now that babel-eslint will actually inform us when a flow type is undefined, which is what you said above, but I didn't understand, @sebmck.

How do I inform babel-eslint of my types that are defined in external files?

Also, I get the 'no-undef' error on functions that return ReactElement, does that have to do with eslint-plugin-react?

**Comment by hzoo** [comment]
@murphyrandle yeah we added support for flow types.

I think there's two ways to solve the external files. One is to import the types at the top like `import type Course from './external-file';`. The other is to define `globals` in your `.eslintrc`? ESLint also has environements with `"env": node` that import pre-defined globals.

Yeah if you are using react/jsx and are getting errors, our current recommendation is to use the plugin you mentioned.

**Comment by murphyrandle** [comment]
okay, I'll take a look at importing types, thanks @hzoo.

Yes, here's an example of the react error:

```javascript
var React = require('react')

module.exports = React.createClass({
  displayName: 'Test',
  render(): ReactElement {
    return (
      <div></div>
    )
  }
})
```

```
example.jsx
  5:12  error  "ReactElement" is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by hzoo** [comment]
Yeah for that you can just add it to your globals.
If `ReactElement` is part of the react package then you could do `import type {React} from 'react';` but not sure if that works.

**Comment by murphyrandle** [comment]
Okay, for now I'm defining them as globals. Thanks, @hzoo.

**Comment by danez** [comment]
since flow 0.22 you need to import ReactElement with
```
import { Element } from 'react';
```
